HeterogeneousCore/AlpakaInterface/test/alpaka/testBuffer.dev.cc

```diff
@@ -0,0 +1,91 @@
+#include <alpaka/alpaka.hpp>
+
+#define CATCH_CONFIG_MAIN
+#include <catch.hpp>
+
+#include "FWCore/Utilities/interface/stringize.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/memory.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/workdivision.h"
+
+// each test binary is built for a single Alpaka backend
+using namespace ALPAKA_ACCELERATOR_NAMESPACE;
+
+namespace {
+  constexpr size_t SIZE = 32;
+
+  void testDeviceSideError(Device const& device) {
+    auto queue = Queue(device);
+    auto buf_h = cms::alpakatools::make_host_buffer<int[]>(queue, SIZE);
+    auto buf_d = cms::alpakatools::make_device_buffer<int[]>(queue, SIZE);
+    alpaka::memset(queue, buf_h, 0);
+    alpaka::memcpy(queue, buf_d, buf_h);
+    // On the host device I don't know how to fabricate a device-side
+    // error for which the Alpaka API calls would then throw an
+    // exception. Therefore I just throw the std::runtime_error to
+    // keep the caller side the same for all backends. At least the
+    // test ensures the buffer destructors won't throw exceptions
+    // during the stack unwinding of the thrown runtime_error.
+    if constexpr (std::is_same_v<Device, DevHost>) {
```

Comment:
here I would use 
```suggestion
    if constexpr (std::is_same_v<Device, alpaka::DevCpu>) {
```

because, if we ever enable the TBB backend, we want to test it using the CPU case as well.

(`DevHost` would also do that, of course, but I find it clearer to use `DevHost` only for the host-vs-device case, and `DevCpu` for the cpu backend vs gpu backend case)

Reply:
Thanks, changed